DigitalLearningSolutions.Web.Tests/Controllers/ForgotPassword/ForgotPasswordControllerTests.cs

```diff
@@ -1,45 +1,87 @@
 ﻿namespace DigitalLearningSolutions.Web.Tests.Controllers.ForgotPassword
 {
+    using System.Security.Claims;
     using DigitalLearningSolutions.Data.Exceptions;
     using DigitalLearningSolutions.Data.Services;
+    using DigitalLearningSolutions.Data.Tests.Helpers;
     using DigitalLearningSolutions.Web.Controllers;
     using DigitalLearningSolutions.Web.ViewModels.ForgotPassword;
     using FakeItEasy;
-    using FluentAssertions;
     using FluentAssertions.AspNetCore.Mvc;
+    using Microsoft.AspNetCore.Http;
+    using Microsoft.AspNetCore.Mvc;
     using NUnit.Framework;
 
     internal class ForgotPasswordControllerTests
     {
         private ForgotPasswordController controller;
+        private ISession httpContextSession;
         private IPasswordResetService passwordResetService;
 
         [SetUp]
         public void SetUp()
         {
             passwordResetService = A.Fake<IPasswordResetService>();
 
-            controller = new ForgotPasswordController(passwordResetService);
+            // Set up unauthenticated user
+            var user = new ClaimsPrincipal(new ClaimsIdentity());
+            httpContextSession = new MockHttpContextSession();
+            controller = new ForgotPasswordController(passwordResetService)
+            {
+                ControllerContext = new ControllerContext
+                {
+                    HttpContext = new DefaultHttpContext
+                    {
+                        User = user,
+                        Session = httpContextSession
+                    }
+                }
+            };
         }
 
         [Test]
-        public void Render_basic_form_when_there_is_no_error_model()
+        public void Index_should_render_if_user_is_unauthenticated()
         {
             // When
-            var result = controller.Index(model: null);
+            var result = controller.Index();
 
             // Then
             result.Should().BeViewResult().WithDefaultViewName();
         }
 
+        [Test]
+        public void Index_should_redirect_if_user_is_authenticated()
+        {
+            // Given user is authenticated
+            var user = new ClaimsPrincipal(new ClaimsIdentity("mock"));
```

Comment:
Sometimes when commenting is required in code to explain what the code is doing, a small restructure of the code can help. In this case, it might be clearer to have a function like getControllerWithMockUser(), so the test is more to the point. Then, if anyone wants to see how the mock user is set up, they can simply navigate into the private function. :)
e.g.

`// Given
controller = getControllerWithMockUser()

// When
var result = controller.Index();

// Then
result.Should().BeRedirectToActionResult().WithControllerName("Home").WithActionName("Index");`